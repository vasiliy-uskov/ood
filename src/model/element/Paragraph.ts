export class Paragraph {
	public readonly text: string;
}