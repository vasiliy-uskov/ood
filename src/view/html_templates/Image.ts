import {Size} from "../../utils/Size";

export type ImageProps = {
	readonly src: string,
	readonly size: Size
};

export function Image(props: ImageProps) {
	return `<img src="${props.src}" width="${props.size.width}" height="${props.size.height}" alt="" />`;
}