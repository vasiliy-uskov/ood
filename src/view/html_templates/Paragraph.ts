import {escapeHtmlText} from "../../utils/utils";

export type ParagraphProps = {
	readonly text: string
};

export function Paragraph(props: ParagraphProps) {
	return `<p>${escapeHtmlText(props.text)}</p>`
}