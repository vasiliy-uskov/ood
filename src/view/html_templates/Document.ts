import {escapeHtmlText} from "../../utils/utils";
import {Image, ImageProps} from "./Image";
import {Paragraph, ParagraphProps} from "./Paragraph";

export enum ContentType {
	PARAGRAPH = 'paragraph',
	IMAGE = 'image',
}

export type ParagraphDocumentItem = {
	type: ContentType.PARAGRAPH,
	props: ParagraphProps,
}

export type ImageDocumentItem = {
	type: ContentType.IMAGE,
	props: ImageProps,
}

export type DocumentContent = ReadonlyArray<(ParagraphDocumentItem|ImageDocumentItem)>;

export type DocumentProps = {
	title: string,
	content: DocumentContent,
}

export function Document(props: DocumentProps) {
	return `<!DOCTYPE html>
<html lang="en">
<head>
	<meta charset="UTF-8">
	<title>${escapeHtmlText(props.title)}</title>
</head>
<body>
	${props.content.map(item => {
		switch (item.type) {
			case ContentType.IMAGE:
				return Image(item.props);
			case ContentType.PARAGRAPH:
				return Paragraph(item.props);
			default:
				throw new Error("Undefined content type")
		}
	}).join("")}
</body>
</html>`;
}