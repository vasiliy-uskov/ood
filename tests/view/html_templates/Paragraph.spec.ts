import {Paragraph} from "../../../src/view/html_templates/Paragraph";

function expandParagraph(text: string) {
	expect(Paragraph({text})).toMatchSnapshot();
}

it('expand empty paragraph', () => {
	expandParagraph('');
});

it('expand not empty paragraph', () => {
	expandParagraph('Some text');
});

it('escape html text', () => {
	expandParagraph(`<p>&nbsp;</p>`);
});