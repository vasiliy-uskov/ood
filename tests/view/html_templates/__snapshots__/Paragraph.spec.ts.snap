// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`escape html text 1`] = `"<p>&lt;p&gt;&amp;nbsp;&lt;/p&gt;</p>"`;

exports[`expand empty paragraph 1`] = `"<p></p>"`;

exports[`expand not empty paragraph 1`] = `"<p>Some text</p>"`;
